mapie/mondrian.py

```diff
@@ -0,0 +1,390 @@
+            method that may be specific to the Mapie estimator used
+
+        Returns
+        -------
+        y_pred : NDArray of shape (n_samples,) or (n_samples, n_outputs)
+            The predicted values
+        y_pss : NDArray of shape (n_samples, n_outputs, n_alpha)
+        """
+
+        check_is_fitted(self, self.fit_attributes)
+        X = cast(NDArray, X)
+        groups = self._check_groups_predict(X, groups)
+        alpha_np = cast(NDArray, check_alpha(alpha))
+        if alpha_np is None and self.mapie_estimator.estimator is not None:
+            return self.mapie_estimator.estimator.predict(
+                X, **predict_params
+            )
+        else:
+            if isinstance(self.mapie_estimator, MapieClassifier):
+                y_pred = np.empty(
+                    (X.shape[0], )
+                )
+                y_pss = np.empty(
+                    (
+                        X.shape[0],
+                        self.n_classes,
+                        len(alpha_np)
+                    )
+                )
+            else:
+                y_pred = np.empty((X.shape[0],))
+                y_pss = np.empty((X.shape[0], 2, len(alpha_np)))
+            unique_groups = np.unique(groups)
+            for i, group in enumerate(unique_groups):
+                indices_groups = np.argwhere(groups == group)[:, 0]
+                X_g = X[indices_groups]
+                y_pred_g, y_pss_g = self.mapie_estimators[group].predict(
+                    X_g, alpha=alpha_np, **predict_params
+                )
+                y_pred[indices_groups] = y_pred_g
+                y_pss[indices_groups] = y_pss_g
+
+            return y_pred, y_pss
+
+    def _check_cv(self):
+        """
+        Check that the underlying Mapie estimator uses cv='prefit'
+
+        Raises
+        ------
+        ValueError
+            If the underlying Mapie estimator does not use cv='prefit'
+        """
+        if not self.mapie_estimator.cv == "prefit":
+            raise ValueError(
+                "Mondrian can only be used if the underlying Mapie" +
+                "estimator uses cv='prefit'."
+            )
+
+    def _check_groups_fit(self, X: NDArray, groups: NDArray):
+        """Check that each group is defined by an integer and check that there
+        are at least 2 individuals per group
+
+        Parameters
+        ----------
+        X : NDArray of shape (n_samples, n_features)
+            The input data
+        groups : NDArray of shape (n_samples,)
+
+        Raises
+        ------
+        ValueError
+            If the groups are not defined by integers
+            If there is less than 2 individuals per group
+            If the number of individuals in the groups is not equal to the
+            number of rows in X
+        """
+        if not np.issubdtype(groups.dtype, np.integer):
+            raise ValueError("The groups must be defined by integers")
```

Comment:
Like `_check_group_length`, this test must be performed in the fit method as well as in the predict method.

Reply:
We already check that the groups of the in the predict are in the fit